Core/src/Propagator/StraightLineStepper.cpp

```diff
@@ -16,75 +16,46 @@ namespace Acts {
 std::tuple<BoundTrackParameters, BoundMatrix, double>
 StraightLineStepper::boundState(State& state, const Surface& surface,
                                 bool transportCov) const {
-  FreeVector parameters;
-  parameters[eFreePos0] = state.pos[ePos0];
-  parameters[eFreePos1] = state.pos[ePos1];
-  parameters[eFreePos2] = state.pos[ePos2];
-  parameters[eFreeTime] = state.t;
-  parameters[eFreeDir0] = state.dir[eMom0];
-  parameters[eFreeDir1] = state.dir[eMom1];
-  parameters[eFreeDir2] = state.dir[eMom2];
-  parameters[eFreeQOverP] = state.q / state.p;
   return detail::boundState(
       state.geoContext, state.cov, state.jacobian, state.jacTransport,
-      state.derivative, state.jacToGlobal, parameters,
+      state.derivative, state.jacToGlobal, state.pars,
       state.covTransport && transportCov, state.pathAccumulated, surface);
 }
 
 std::tuple<CurvilinearTrackParameters, BoundMatrix, double>
 StraightLineStepper::curvilinearState(State& state, bool transportCov) const {
-  FreeVector parameters;
-  parameters[eFreePos0] = state.pos[ePos0];
-  parameters[eFreePos1] = state.pos[ePos1];
-  parameters[eFreePos2] = state.pos[ePos2];
-  parameters[eFreeTime] = state.t;
-  parameters[eFreeDir0] = state.dir[eMom0];
-  parameters[eFreeDir1] = state.dir[eMom1];
-  parameters[eFreeDir2] = state.dir[eMom2];
-  parameters[eFreeQOverP] = state.q / state.p;
   return detail::curvilinearState(
       state.cov, state.jacobian, state.jacTransport, state.derivative,
-      state.jacToGlobal, parameters, state.covTransport && transportCov,
+      state.jacToGlobal, state.pars, state.covTransport && transportCov,
       state.pathAccumulated);
 }
 
 void StraightLineStepper::update(State& state, const FreeVector& parameters,
                                  const Covariance& covariance) const {
-  state.pos = parameters.template segment<3>(eFreePos0);
-  state.dir = parameters.template segment<3>(eFreeDir0).normalized();
-  state.p = std::abs(1. / parameters[eFreeQOverP]);
-  state.t = parameters[eFreeTime];
+  state.pars = parameters;
   state.cov = covariance;
 }
 
 void StraightLineStepper::update(State& state, const Vector3D& uposition,
                                  const Vector3D& udirection, double up,
                                  double time) const {
-  state.pos = uposition;
-  state.dir = udirection.normalized();
-  state.p = up;
-  state.t = time;
+  state.pars.template segment<3>(eFreePos0) = uposition;
+  state.pars.template segment<3>(eFreeDir0) = udirection;
+  state.pars[eFreeTime] = time;
+  state.pars[eFreeQOverP] = state.q / up;
 }
 
 void StraightLineStepper::covarianceTransport(State& state) const {
   detail::covarianceTransport(state.cov, state.jacobian, state.jacTransport,
-                              state.derivative, state.jacToGlobal, state.dir);
+                              state.derivative, state.jacToGlobal,
+                              state.pars.template segment<3>(eFreeDir0));
```

Comment:
```suggestion
                              direction(state);
```

Reply:
I think it's good as it is for the moment, `direction(state)` doesn't do anything else.

Reply:
Indeed. I just prefer calling functions instead of using the data from the state directly, e.g. for debugging.